### PR TITLE
feat: Add `OBSERVER` mode to SQL repl along with SQL primer / cookbook

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
While trying to understand the state of an IOx server, I want to be able to run queries on system tables across databases (so I can have a consolidated view of the system)

# Changes
Follow on to https://github.com/influxdata/influxdb_iox/pull/1332, and based on code from https://github.com/influxdata/influxdb_iox/pull/1117

1. Add the observer mode to SQL repl, with tests
2. Add a quickie "Common queries" document to the admin guide with a cheatsheet of common queries

# Examples of use:

```
OBSERVER> observer;
Preparing local views of remote system tables
Loading system tables from 49 databases
...................................................................................................................................................
 Completed in 3.654190485s
You are now in Observer mode.

SQL commands in this mode run against a cached unified view of
remote system tables in all remote databases.

To see the unified tables available to you, try running
SHOW TABLES;

To reload the most recent version of the database system tables, run
OBSERVER;

Example SQL to show the total estimated storage size by database:

SELECT database_name, storage, count(*) as num_chunks,
  sum(estimated_bytes)/1024/1024 as estimated_mb
FROM chunks
GROUP BY database_name, storage
ORDER BY estimated_mb desc;

OBSERVER> SELECT
  database_name, count(*) as num_chunks,
  sum(estimated_bytes)/1024/1024 as estimated_mb
FROM chunks
GROUP BY database_name
ORDER BY estimated_mb desc
LIMIT 20;

+-----------------------------------+------------+--------------+
| database_name                     | num_chunks | estimated_mb |
+-----------------------------------+------------+--------------+
| 844910ece80be8bc_eaec8df57a81a1e9 | 312        | 979          |
| 844910ece80be8bc_3c0bd4c89186ca89 | 371        | 884          |
| 844910ece80be8bc_64881f0f8fd1653c | 368        | 261          |
| 844910ece80be8bc_frontendservices | 60         | 49           |
| 844910ece80be8bc_05d1e95653672000 | 73         | 49           |
| 844910ece80be8bc_7caaf9613b16e7c2 | 31         | 49           |
| 844910ece80be8bc_b79491dc43e2c47e | 81         | 48           |
| 844910ece80be8bc_istio            | 1151       | 47           |
| 844910ece80be8bc_6f8925e75446c183 | 1881       | 46           |
| 7f4b06e9112d7bc8_system%5Fusage   | 14         | 43           |
| 844910ece80be8bc_apps             | 344        | 42           |
| InfluxData_%5Ftasks               | 6          | 32           |
| 844910ece80be8bc_5d5a6ade9b665dfc | 292        | 25           |
| 844910ece80be8bc_1f8d706e48cf8e70 | 12         | 20           |
| 844910ece80be8bc_81c1fce8c36339dc | 33         | 19           |
| 844910ece80be8bc_4bed41a6ff7f0ee0 | 152        | 16           |
| 844910ece80be8bc_infra            | 446        | 12           |
| 844910ece80be8bc_dfeaf11d9d194efd | 315        | 10           |
| 844910ece80be8bc_84cc8db6499a3d11 | 10         | 10           |
| 844910ece80be8bc_a44813fabdadbc9d | 149        | 5            |
+-----------------------------------+------------+--------------+

Query execution complete in 12.862755ms
```

### CODE REVIEW

File: docs/sql.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# InfluxDB SQL Repl Tips and Tricks
```
Comment (alamb) [inline docs/sql.md:1]: FYI @mkmik  here is my SQL cheat sheet / primer for system table queries

File: docs/sql.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+### Total row count by partition and table
+
+```sql
+SELECT database_name, partition_key, table_name, max(count) as total_rows
+FROM columns
+GROUP BY database_name, partition_key, table_name
+ORDER BY total_rows DESC
+LIMIT 20;
+```
```
Comment (NGA-TRAN) [inline docs/sql.md:214]: I like this doc

**Review by NGA-TRAN — APPROVED** [review]
You really adding more and more easy-to-use tools for ourselves and customers. I like this a lot.

My question for the future work is when we add security features such as user roles, authentication, and authorization, we will limit what DBs they can see, right?

**Comment by pauldix** [comment]
@NGA-TRAN specific access control is out of scope for IOx. We'll have something like JWT authentication, but that'll be an all or nothing access.

File: docs/sql.md (hunk 1 above)
Comment (pauldix) [reply docs/sql.md:1]: This doc is awesome, thank you!

**Review by crepererum — APPROVED** [review]

**Pull request merged**